plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -160,11 +160,16 @@ func NodeRules() []rbacv1.PolicyRule {
 		volAttachRule := rbacv1helpers.NewRule("get").Groups(storageGroup).Resources("volumeattachments").RuleOrDie()
 		nodePolicyRules = append(nodePolicyRules, volAttachRule)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletPluginsWatcher) {
+		csiNodeInfoRule := rbacv1helpers.NewRule("get", "watch", "list", "update", "patch", "create").Groups("csi.storage.k8s.io").Resources("csinodeinfos").RuleOrDie()
+		nodePolicyRules = append(nodePolicyRules, csiNodeInfoRule)
```

Comment:
This lets any node update the CSI info for any other node. I expected nodes to only be able to update their own CSI info. That is typically accomplished via changes in the node authorizer and noderestriction admission plugins. 

@mtaufen for pointers from your recent node lease change

@mikedanese fyi

Reply:
Should the Node Restriction rule be added *in addition to* the RBAC rule here, in case NodeRestriction admission plugin is disabled?

Reply:
Looks like the authorizer uses internal APIs for objects it authorizes ([link](https://github.com/kubernetes/kubernetes/blob/ef741c3038bb26326daed17ac722cce8843b8f0d/plugin/pkg/auth/authorizer/node/node_authorizer.go#L30)). For `CSINodeInfo` this has to be external, i.e. "k8s.io/csi-api/pkg/apis/csi/v1alpha1". Would this cause any issues?

Reply:
No, only resource and API group are required to authorize. Version is not used.

Reply:
> Are there any special considerations if the object is a CRD?

If the authorization and admission restriction is purely based on the name of the incoming object, then no. If watching the existing objects with an informer was required, that might be more involved, but iirc, the intent was to have the name of the csi info object match the node name, right?

Reply:
> but iirc, the intent was to have the name of the csi info object match the node name, right?
Yep that's correct, and we don't use an informer here.

For the external provisioner, it's sufficient to have just the RBAC read rules, since it needs to read from all CSINodeInfo objects, right?